src/astronomer/flask_appbuilder/security.py

```diff
@@ -186,6 +186,16 @@ def before_request(self):
             self.get_session.commit()
             if not login_user(user):
                 raise RuntimeError("Error logging user in!")
+            role_names = []
+            for role in user.roles:
+                role_names.append(role.name)
+            session["roles"] = role_names
+        else:
+            session_roles = session['roles']
+            claim_roles = claims['roles']
+            if set(session_roles) != set(claim_roles):
+                logout_user()
+                return redirect('/')
```

Comment:
Rather than logging out and redirecting we can just update them directly:

```suggestion
                self.manage_user_roles(current_user, claims['roles'])
```

Should do it I think

Reply:
This had a weird side affect I wanted to avoid. Essentially If a non-admin was on a page they were not allowed to view (say the variables page) it would navigate them to our whoops something went wrong page. That page has a button on it that performs a javascript window.history.back(); operation. Which sends the user back to the original page and it appears they still have admin permissions. Essentially putting them in an endless loop of malfunction until they decide to reload the page.

Reply:
Hmmm, curious, I though most of that was handled in Airflow by it redirecting to the Airflow home page (which is not `/`, but has the deployment name in it and then showing a flash message.)

Ah, in which case instead of sending them to `/` You should use redirect(url_for('Airflow.home')) -- possibly also with a flash to tell them _why_ they ended up on this page not where they expected to be.

But an even better fix: Update our ingress config so that 403 pages are handled properly (I think right now it even redircts to a `/error` URL so you can't even see _what_ the previous URL was.)?

Reply:
My main reason for wanting to avoid redirecting to `/` is that when you click Open Airflow it's in a new window, so this would just mean you'd have two windows open with the Astro UI :)

Reply:
Much agreed this is a better flow. I have updated the PR to go along the redirect to airflow homepage route. I updated the PR desc gif as well if you want to see it in action.